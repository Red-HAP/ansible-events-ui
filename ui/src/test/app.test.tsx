import * as React from 'react';
import { act } from 'react-dom/test-utils';
import App from '@app/index';
import { mount, shallow } from 'enzyme';
import { Button } from '@patternfly/react-core';
import {MemoryRouter} from "react-router";
import fetchMock from "jest-fetch-mock";
import {AppLayout} from "@app/AppLayout/AppLayout";
import {AppRoutes} from "@app/routes";

const ComponentWrapper = ({ children }) => (
    <MemoryRouter initialEntries={['/eda/dashboard']}>{children}</MemoryRouter>
);
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/eda/dashboard"
  })
}));

describe('App tests', () => {
  test('should render default App component', () => {
    const view = shallow(<App />);
    expect(view).toMatchSnapshot();
  });

  it('should render a nav-toggle button', () => {
    const wrapper = mount(<App />);
    const button = wrapper.find(Button);
    expect(button.exists()).toBe(true);
  });

  it('should hide the sidebar on smaller viewports', async () => {
    fetchMock.mockResponse(JSON.stringify({}))
    Object.defineProperty(window, 'innerWidth', { writable: true, configurable: true, value: 200 });
    let wrapper;
    await act(async () => {
      wrapper = mount(
        <ComponentWrapper>
          <AppLayout>
            <AppRoutes/>
          </AppLayout>
        </ComponentWrapper>
      );
    });
    await act(async () => {
      wrapper.update();
    });
    window.dispatchEvent(new Event('resize'));
    await act(async () => {
      wrapper.update();
    });
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed')).toBeTruthy();
  });

  it('should hide the sidebar when clicking the nav-toggle button', () => {
    Object.defineProperty(window, 'innerWidth', { writable: true, configurable: true, value: 1200 });
    const wrapper = mount(
      <ComponentWrapper>
        <AppLayout>
          <AppRoutes/>
        </AppLayout>
      </ComponentWrapper>);
    window.dispatchEvent(new Event('resize'));
    const button = wrapper.find('#nav-toggle').hostNodes();
    button.simulate('click');
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed')).toBeTruthy();
    button.simulate('click');
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed')).toBeFalsy;
  });
});
