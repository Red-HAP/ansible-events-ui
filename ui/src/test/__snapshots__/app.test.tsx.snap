// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<Fragment>
  <Memo(l) />
  <IntlProvider
    defaultFormats={Object {}}
    defaultLocale="en"
    fallbackOnEmptyString={true}
    formats={Object {}}
    locale="en"
    messages={Object {}}
    onError={[Function]}
    onWarn={[Function]}
    textComponent={Symbol(react.fragment)}
  >
    <BrowserRouter
      basename="/eda"
    >
      <Switch>
        <Route
          exact={true}
          path="/"
        >
          <Login />
        </Route>
        <Route>
          <AppLayout>
            <AppRoutes />
          </AppLayout>
        </Route>
      </Switch>
    </BrowserRouter>
  </IntlProvider>
</Fragment>
`;
